### PR TITLE
Update e2e tests that depends on the HCA

### PR DESCRIPTION
**Pull request by erikphansen**
## Description
Fix tests that use the HCA and don't know about:
- the new ID Page at the front of the form for logged in users
- the new `enrollment_system` endpoint that intro page hits when users are logged it

## Testing done
Local

## Acceptance criteria
- [ ] tests work locally and on Jenkins

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/forms/tests/save-in-progress/01-sip-autosave.e2e.spec.js [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
-const Timeouts = require('../../../testing/e2e/timeouts.js');
-const HcaHelpers = require('../../../../applications/hca/tests/hca-helpers.js');
+const E2eHelpers = require('platform/testing/e2e/helpers');
+const Timeouts = require('platform/testing/e2e/timeouts.js');
+const HcaHelpers = require('applications/hca/tests/hca-helpers.js');
 
 module.exports = E2eHelpers.createE2eTest(client => {
   const url = `${E2eHelpers.baseUrl}/health-care/apply/application`;
   const token = HcaHelpers.initSaveInProgressMock(url, client);
+  HcaHelpers.initEnrollmentStatusMock(token);
```
Comment (erikphansen) [inline src/platform/forms/tests/save-in-progress/01-sip-autosave.e2e.spec.js]: This test is using the HCA to test autosave functionality. To make this test pass locally (and in CI when the HCA changes are pushed to prod) we need to mock the `enrollment_status` endpoint that the introduction page will depend on.

File: src/platform/forms/tests/save-in-progress/01-sip-load-fail.e2e.spec.js [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
-const Timeouts = require('../../../testing/e2e/timeouts.js');
-const HcaHelpers = require('../../../../applications/hca/tests/hca-helpers.js');
+const E2eHelpers = require('platform/testing/e2e/helpers');
+const Timeouts = require('platform/testing/e2e/timeouts.js');
+const HcaHelpers = require('applications/hca/tests/hca-helpers.js');
 
 module.exports = E2eHelpers.createE2eTest(client => {
   const url = `${E2eHelpers.baseUrl}/health-care/apply/application`;
   const token = HcaHelpers.initSaveInProgressMock(url, client);
+  HcaHelpers.initEnrollmentStatusMock(token);
```
Comment (erikphansen) [inline src/platform/forms/tests/save-in-progress/01-sip-load-fail.e2e.spec.js]: This test is using the HCA to test autosave functionality. To make this test pass locally (and in CI when the HCA changes are pushed to prod) we need to mock the `enrollment_status` endpoint that the introduction page will depend on.

File: src/applications/hca/tests/hca-helpers.js [hunk 3]
```diff
@@ -368,6 +377,19 @@ function initApplicationSubmitMock() {
   });
 }
 
+function initEnrollmentStatusMock(token = null) {
```
Comment (erikphansen) [inline src/applications/hca/tests/hca-helpers.js]: This mock of the `enrollment_status` endpoint mocks a user who is in MVI but who has not applied for health care.

**Review by rianfowler — APPROVED** [review]

**Review by tlei123 — COMMENTED** [review]
LGTM

**Pull request merged**